src/formatDate.js

```diff
@@ -50,7 +50,45 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newSeparator = toFormat.slice(-1);
+  const dateArr = date.split(fromFormat[3]);
+  const newDate = [];
+
+  let year, month, day;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].includes('Y')) {
+      year = dateArr[i];
+    }
+
+    if (fromFormat[i].includes('M')) {
+      month = dateArr[i];
+    }
+
+    if (fromFormat[i].includes('D')) {
+      day = dateArr[i];
+    }
+  }
+
+  for (const format of toFormat) {
+    if (format.includes('Y')) {
+      if (format.length < year.length) {
+        newDate.push(year.slice(2));
+      } else {
+        newDate.push(year.padStart(format.length, year >= 30 ? '19' : '20'));
```

Comment:
it's cool that you use new func, but this func is using where we need to duplicate something many times, in this case you just need to add 19 or 20 in start of string. And it may lead to difficulty understanding what is written.